### PR TITLE
Add `dcos_service_account_secret` data resource

### PR DESCRIPTION
**Pull request by wavesoft**
This commit introduces the `dcos_service_account_secret` data resource that can be used to create valid contents for the service account secret. Example:

```hcl
# Create a private key using an external provider (eg. tls)
resource "tls_private_key" "service_account_key" {
  algorithm = "RSA"
  rsa_bits  = "2048"
}

# Pass the private key and the user account to the resource
data "dcos_service_account_secret" "service_account" {
    uid            = "user-login"
    private_key    = "${tls_private_key.service_account_key.private_key_pem}"
}

# Handle the contents (eg. upload to a secret)
resource "dcos_security_secret" "service_account_secret" {
  path = "my-service/service-account"
  value = "${data.dcos_service_account_secret.contents}"
}
```

### CODE REVIEW

File: dcos/data_source_dcos_service_account_secret.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+				Description: "The user ID",
+			},
+			"login_endpoint": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Default:     "https://leader.mesos/acs/api/v1/auth/login",
+				Description: "Customize the login_endpoint parameter",
+			},
+			"contents": {
```
Comment (fatz) [inline dcos/data_source_dcos_service_account_secret.go]: what about naming this `"json"` ?

File: dcos/provider.go [hunk 2]
```diff
@@ -72,13 +72,14 @@ func Provider() terraform.ResourceProvider {
...
-			"dcos_token":           dataSourceDcosToken(),
-			"dcos_version":         dataSourceDcosVersion(),
+			"dcos_base_url":               dataSourceDcosBaseURL(),
+			"dcos_job":                    dataSourceDcosJob(),
+			"dcos_package_config":         dataSourceDcosPackageConfig(),
+			"dcos_package_version":        dataSourceDcosPackageVersion(),
+			"dcos_service":                dataSourceDcosService(),
+			"dcos_token":                  dataSourceDcosToken(),
+			"dcos_version":                dataSourceDcosVersion(),
+			"dcos_service_account_secret": dataSourceDcosServiceAccountSecret(),
```
Comment (fatz) [inline dcos/provider.go]: `dcos_security_secret_service_account_secret` as of `dcos security secrets create-sa-secret`

File: dcos/data_source_dcos_service_account_secret.go (hunk 1 above)
Comment (wavesoft) [reply dcos/data_source_dcos_service_account_secret.go]: I was thinking that this should be treated as an opaque variable, and hence the content-agnostic `content` variable.

The user shouldn't care if the contents are JSON or anything that we might consider in the future 🤔

File: dcos/provider.go (hunk 2 above)
Comment (wavesoft) [reply dcos/provider.go]: Yup, that sounds good 👍

File: dcos/data_source_dcos_service_account_secret.go (hunk 1 above)
Comment (fatz) [reply dcos/data_source_dcos_service_account_secret.go]: ok that also makes sense to me

**Pull request merged**